src/api/mod.rs

```diff
@@ -0,0 +1,68 @@
+use std::collections::HashMap;
+
+use crate::utility::{URI, ENDPOINT_CREDITS, ZBError};
+
+pub struct ZeroBounce<'key> {
+    api_key: &'key str,
+    client: reqwest::blocking::Client
+}
+
+impl ZeroBounce<'_> {
+    pub fn new(api_key: &str, client: reqwest::blocking::Client) -> ZeroBounce {
+        ZeroBounce {api_key, client}
+    }
+
+    fn get_credits_from_string(string_value: String) -> Result<i64, ZBError> {
+
+        let parsed_value = serde_json::from_str::<serde_json::Value>(string_value.as_ref())
+            .map_err(ZBError::JsonError)?;
+
+        parsed_value
+            .get("Credits")
+            .and_then(serde_json::Value::as_str)
+            .map(str::parse::<i64>)
+            .ok_or(ZBError::explicit("credits value not in json"))?
+            .map_err(ZBError::IntParseError)
+    }
+
+    pub fn get_credits(&self) -> Result<i64, ZBError> {
+        
+        let url_string = URI.to_owned() + ENDPOINT_CREDITS;
+        
+        let mut query_args = HashMap::<&str, &str>::new();
+        query_args.insert("api_key", self.api_key);
+
+        let response = self.client.get(url_string)
+            .query(&query_args)
+            .send()
+            .map_err(ZBError::RequestError)?;
+
+        let response_ok = response.status().is_success();
+
+        let response_content = response
+            .text()
+            .map_err(ZBError::RequestError)?;
+
+        if !response_ok {
+            return Err(ZBError::explicit(response_content.as_str()));
+        }
+
+        Self::get_credits_from_string(response_content)
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn forbidden_integration_test() {
+        let key = "c1b432f97c0145d0aca98b2f80b044da";
+        let credits = ZeroBounce::new(key, reqwest::blocking::Client::new())
```

Comment:
Added mocking in [another PR](https://github.com/zerobounce/zerobounce-rust-api/pull/2/files#diff-bafccc544f265dceb258113b9b604c7e58d5b80b4979d5ef16280dd65597a63e) (because.. reasons).

Is this enough to not be mocked?

Reply:
possibly; I've reviewed the latter PR as well and you split the method into get request and parse result and only use parse result in the tests; can you look into the http mock library I linked and see if that allows you to test the whole implementation?

Reply:
![image](https://user-images.githubusercontent.com/128589427/230348709-5033aaa8-19bd-4d53-b48e-ef2583d20501.png)

Seems to be the same situation: the library creates an API that the requests must be directed towards.